(feat): Autosave using localStorage (#6864)
plone/volto: docs/source/user-manual/blocks.md

The blocks editor allows you to add, modify, reorder, and delete blocks given your requirements.
Blocks provide the user the ability to display content in a specific way, although they can also define behavior and have specific features.

(autosave-label)=

## Autosave

Volto provides an autosave mechanism that automatically captures your ongoing edits as you work on a page or content item.

Changes made in the blocks editor are periodically saved in the background to prevent data loss, even if the user navigates away or the browser unexpectedly closes.

Autosave stores changes locally in your browser using [`localStorage`](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage).
These changes remain available until you explicitly click the {guilabel}`Save` button <img alt="Save icon" src="../_static/user-manual/blocks/save.svg" class="inline">, at which point they are persisted to the server and the local storage is cleared.

Watch the video below for a quick demonstration.

> ```{video} ../_static/user-manual/blocks/autosave-local.mp4
> :alt: Autosave
> ```

Reviewer: Sorry, I goofed on the videos. They should have this wrapper.

`````suggestion
````{only} not text
```{video} ../_static/user-manual/blocks/autosave-local.mp4
:alt: Autosave
```
````
`````

### Autosave and edit conflicts

If another user edits and saves the same page while you're still working on it, Volto detects the conflict between your autosaved version and the newer version saved on the server.

In this case, when you return to the editor, Volto displays a message informing you that the page content has changed, but a previously autosaved version still exists.

You will be prompted to choose whether you want to restore your autosaved version or continue with the newer version from the server.
After making your choice, you can also undo that action.

> ```{video} ../_static/user-manual/blocks/autosave-remote.mp4

Reviewer: `````suggestion
````{only} not text
```{video} ../_static/user-manual/blocks/autosave-remote.mp4
```
````
`````

Alignment
: Options for alignment include left, right, center, and full width.

## Autosave

> Volto provides an **autosave** mechanism that automatically captures your ongoing edits as you work on a page or content item.

Reviewer: ```suggestion
Volto provides an autosave mechanism that automatically captures your ongoing edits as you work on a page or content item.
```

Changes made in the blocks editor are periodically saved in the background to prevent data loss, even if the user navigates away or the browser unexpectedly closes.

> `````{note}
> Autosave stores changes locally in your browser using `localStorage`. These changes remain available until you explicitly click the **Save** button, at which point they are persisted to the server and the local storage is cleared.
> 
> Watch the video below for a quick demonstration.
> 
> ````{only} not text
> ```{video} ../_static/user-manual/blocks/autosave-local.mp4
> :alt: Autosave
> `````

Reviewer: Let's simplify the markup. A note isn't needed here, as it continues the previous paragraph explanation of what is going on and how to use it.

Also copy `packages/volto/src/icons/save.svg` to `docs/source/_static/user-manual/blocks/save.svg`

````suggestion
Autosave stores changes locally in your browser using [`localStorage`](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage).
These changes remain available until you explicitly click the {guilabel}`Save` button <img alt="Save icon" src="../_static/user-manual/blocks/save.svg" class="inline">, at which point they are persisted to the server and the local storage is cleared.

Watch the video below for a quick demonstration.

```{video} ../_static/user-manual/blocks/autosave-local.mp4
:alt: Autosave
```
````

### Autosave and edit conflicts

> If another user edits and saves the same page while you're still working on it, Volto detects the conflict between your **autosaved version** and the **newer version saved on the server**.

Reviewer: ```suggestion
If another user edits and saves the same page while you're still working on it, Volto detects the conflict between your autosaved version and the newer version saved on the server.
```

In this case, when you return to the editor, Volto displays a message informing you that the page content has changed, but a previously autosaved version still exists.

> `````{note}

Reviewer: ```suggestion
```

> You will be prompted to choose whether you want to **restore your autosaved version** or continue with the newer version from the server, you can also undo.

Reviewer: ```suggestion
You will be prompted to choose whether you want to restore your autosaved version or continue with the newer version from the server.
After making your choice, you can also undo that action.
```

> ````{only} not text

Reviewer: ```suggestion
```

```{video} ../_static/user-manual/blocks/autosave-remote.mp4
:alt: Autosave conflicts

> `````

Reviewer: ````suggestion
```
````
